demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,508 @@
+        int width;
+        int height;
+        std::vector<std::string> gridMatRowsCols = split(FLAGS_res, 'x');        
+        if (gridMatRowsCols.size() != 2) {
+            throw std::runtime_error("The value of GridMat resolution flag is not valid.");
+        } else {
+            width = std::stoi(gridMatRowsCols[0]);
+            height = std::stoi(gridMatRowsCols[1]);
+        }
+        GridMat gridMat = GridMat(inputImagesCount, presenter, cv::Size(width, height));
+
+        // image, predicted label, isPredictionRight
+        std::list<std::tuple<cv::Mat, std::string, bool>> shownImagesInfo;
+        // ---------------------------------------------------------------------------------------------------
+        
+        // -----------------------------Prepare variables and data for main loop------------------------------
+        double avgFPS = 0;
+        double avgLatency = 0;
+        double latencySum = 0;
+        unsigned framesNum = 0;
+        long long rightPredictionsCount = 0;
+        long long totalPredictionsCount = 0;
+        double accuracy;
+        bool isTestMode = true;
+        int delay = 1;
+        char key = 0;
+        std::size_t nextImageIndex = 0;
+        long long startTickCount = cv::getTickCount();
+        std::condition_variable condVar;
+        std::mutex mutex;
+        std::exception_ptr irCallbackException;
+        
+        std::queue<IRInfo> emptyInferRequests;
+        std::queue<IRInfo> completedInferRequests;
+        for (std::size_t i = 0; i < inferRequests.size(); i++) {
+            emptyInferRequests.push(IRInfo(inferRequests[i], std::vector<IRInfo::IRImage>()));
+        }
+
+        auto startTime = std::chrono::system_clock::now();
+        auto currentTime = std::chrono::system_clock::now();
+        std::chrono::duration<double> elapsedSeconds = currentTime - startTime;
+        // ---------------------------------------------------------------------------------------------------
+        
+
+
+        // ===================================================================================================
+        do {
+            if (irCallbackException) std::rethrow_exception(irCallbackException);
+
+            if (isTestMode && elapsedSeconds.count() >= 10) {
+                isTestMode = false;
+
+                gridMat = GridMat(inputImagesCount, presenter, cv::Size(width, height), cv::Size(16, 9), avgFPS);
+                cv::Size gridMatSize = gridMat.getSize();
+                delay = ((FLAGS_delay == -1)
+                         ? static_cast<int>(avgFPS / (gridMatSize.width * gridMatSize.height) * 1000)
+                         : FLAGS_delay);
+            }
+
+            std::unique_lock<std::mutex> lock(mutex);
+        
+            if (!completedInferRequests.empty()) {
+                auto completedIRInfo = completedInferRequests.front();
+                completedInferRequests.pop();
+                mutex.unlock();
```

Comment:
```suggestion
                lock.unlock();
```
`lock` locked that `mutex` so it should be unlocked via `lock.unlock();`